File: yewonahn/1384.py
import sys
input = sys.stdin.readline

group_num = 1

while True:
    # 마지막 줄에 0 입력하면 종료
    line = input().split()
    if not line:
        break
    n = int(line[0])

    if n == 0:
        break

    names = []
    papers = []

    for i in range(n):
        data = input().split()
        # 첫 번째 이름
        names.append(data[0])
        # 뒤에 메시지
        papers.append(data[1:])

        # 그룹 출력
    if group_num > 1:
        # 그룹 사이 띄움
        print()
    print(f"Group {group_num}")

    is_N = False

    # 종이 주인의 번호 p
    for p in range(n):
        # 종이에 적힌 메시지 순서 q
        for q in range(n-1):
            if papers[p][q] == 'N':
                is_N = True
                N_person_idx = (p-(q+1))%n

                print(f"{names[N_person_idx]} was nasty about {names[p]}")

    if not is_N:
        print("Nobody was nasty")

    group_num += 1

File: lim-jaein/1384.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = 0;
        int group = 0;
        Queue<int[]> q = new ArrayDeque<>();

        while(true) {
            N = Integer.parseInt(br.readLine());
            if (N == 0)
                break;

            String[] crew = new String[N];
            group += 1;

            if (group > 1)
                System.out.println();
            System.out.println("Group " + group);

            for (int i = 0; i < N; i++) {
                String[] input = br.readLine().split("\\s+");

                // 학생리스트 저장
                crew[i] = input[0];
                for (int j = 1; j < N; j++) {
                    if (input[j].equals("N")) {
                        // N-j번째 학생이 i번째에게 나쁜말을 했음
                        q.add(new int[]{i, (i-j+N)%N});
                    }
                }
            }

            if (q.isEmpty()) {
                System.out.println("Nobody was nasty");
            }

            while (!q.isEmpty()) {
                int[] cur = q.poll();
                System.out.println(crew[cur[1]] + " was nasty about " + crew[cur[0]]);
            }
        }

        br.close();
    }
}